File: app/src/main/java/com/djrausch/billtracker/models/BillPaid.java
package com.djrausch.billtracker.models;

import com.google.gson.annotations.SerializedName;

import java.util.Date;
import java.util.UUID;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by white on 7/25/2016.
 */
public class BillPaid extends RealmObject {
    @PrimaryKey
    public String uuid;
    @SerializedName("paid_date")
    public Date date;

    public BillPaid() {

    }

    public BillPaid(Date date) {
        this.uuid = UUID.randomUUID().toString();
        this.date = date;
    }

    @Override
    public String toString() {
        return "BillPaid{" +
                "uuid='" + uuid + '\'' +
                ", date=" + date +
                '}';
    }
}

File: app/src/main/java/com/djrausch/billtracker/models/BillPaid.kt
package com.djrausch.billtracker.models

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import java.util.*

open class BillPaid : RealmObject {
    @PrimaryKey
    open var uuid: String? = null
    open var date: Date? = null

    constructor() {

    }

    constructor(date: Date) {
        this.uuid = UUID.randomUUID().toString()
        this.date = date
    }

    override fun toString(): String {
        return "BillPaid{uuid='$uuid', date='$date'}"
    }

}
